keyboards/kinesis/keymaps/jwon/keymap.c

```diff
@@ -0,0 +1,166 @@
+           _______, _______, _______, _______, _______, _______,
+                    _______, _______, _______, _______,
+           // Left Thumb
+                    _______, _______,
+                             _______,
+           _______, _______, _______,
+
+           // Right Hand
+           _______, _______, _______, _______, _______, _______, _______, _______, _______,
+           _______, KC_NLCK, KC_PEQL, KC_PSLS, KC_PAST, _______,
+           _______, KC_P7,   KC_P8,   KC_P9,   KC_PMNS, _______,
+           _______, KC_P4,   KC_P5,   KC_P6,   KC_PPLS, _______,
+           _______, KC_P1,   KC_P2,   KC_P3,   KC_PENT, _______,
+                    _______, _______, KC_PDOT, KC_PENT,
+           // Right Thumb
+           _______, _______,
+           _______,
+           _______, _______, KC_P0
+    ),
+
+[_QWERTY] = LAYOUT(
+           // Left Hand
+           KC_ESC,  _______, _______, _______, _______, _______, _______, _______, _______,
+           KC_EQL,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,
+           KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,
+           KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,
+           KC_LSFT, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,
+                    KC_GRV,  KC_INS,  KC_LEFT, KC_RGHT,
+           // Left Thumb
+                    KC_LGUI, KC_LALT,
+                             KC_HOME,
+           KC_BSPC, KC_DEL,  KC_END ,
+
+           // Right Hand
+           _______, _______, _______, _______, _______, _______, _______, _______, _______,
+	       KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS,
+	       KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_BSLS,
+	       KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,
+	       KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,
+		            KC_UP,   KC_DOWN, KC_LBRC, KC_RBRC,
+           // Right Thumb
+           KC_RGUI, KC_RCTL,
+           KC_PGUP,
+           KC_PGDN, KC_ENT,  KC_SPC
+    )
+};
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    switch (get_highest_layer(state)) {
+    case _KEYPAD:
+        writePinLow(LED_COMPOSE_PIN);
+        break;
+    case _RAISE_DVORAK:
+        writePinLow(LED_NUM_LOCK_PIN);
+        break;
+    case _QWERTY:
+        writePinLow(LED_SCROLL_LOCK_PIN);
+        break;
+    default: //  for any other layers, or the default layer
+        writePinHigh(LED_NUM_LOCK_PIN);
+        writePinHigh(LED_SCROLL_LOCK_PIN);
+        writePinHigh(LED_COMPOSE_PIN);
+        break;
+    }
+  return state;
+}
+
+bool led_update_user(led_t led_state) {
+    // Disable led_update_kb() so that layer indication code doesn't get overridden.
+    return false;
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        if (get_mods() & MOD_MASK_SHIFT) {
+            writePinLow(LED_CAPS_LOCK_PIN);
+        }
+    } else {
+        writePinHigh(LED_CAPS_LOCK_PIN);
+    }
+    return true;
+}
```

Comment:
You could do this, too:
```suggestion
void matrix_scan_user(void) {
    writePin(LED_CAPS_LOCK_PIN, !(get_mods() & MOD_MASK_SHIFT));
}
```